clang/lib/Basic/Targets/X86.cpp

```diff
@@ -450,6 +450,8 @@ bool X86TargetInfo::handleTargetFeatures(std::vector<std::string> &Features,
       HasFullBFloat16 = true;
     } else if (Feature == "+egpr") {
       HasEGPR = true;
+    } else if (Feature == "+inline-asm-use-gpr32") {
```

Comment:
This should not be implemented as a feature.

Reply:
I want to use `HasInlineAsmUseGPR32` to define a MACRO, Do you know any other reference to implement so?

Reply:
```
  std::string CodeModel = getTargetOpts().CodeModel;
  if (CodeModel == "default")
    CodeModel = "small";
  Builder.defineMacro("__code_model_" + CodeModel + "__");
```

Reply:
`mcmodel` maybe not a good reference here since it's target independent.

Reply:
pls notice that this patch didn't create a real new feature for x86.

Reply:
Will the feature be emitted in IR?

Reply:
> `mcmodel` maybe not a good reference here since it's target independent.

```
  if (Opts.MicrosoftExt && getTriple().getArch() == llvm::Triple::x86) {
    switch (SSELevel) {
    case AVX512F:
    case AVX2:
    case AVX:
    case SSE42:
    case SSE41:
    case SSSE3:
    case SSE3:
    case SSE2:
      Builder.defineMacro("_M_IX86_FP", Twine(2));
      break;
    case SSE1:
      Builder.defineMacro("_M_IX86_FP", Twine(1));
      break;
    default:
      Builder.defineMacro("_M_IX86_FP", Twine(0));
      break;
    }
  }
```

Reply:
> Will the feature be emitted in IR?

yes, I was referring `prefer-no-gather`

Reply:
> > Will the feature be emitted in IR?
> 
> yes, I was referring `prefer-no-gather`

Then it's not expected. `prefer-no-gather` is a real tuning feature.

Reply:
SSELevel relies on feature string in IR attribute as well. it's same situation here.

Reply:
> > > Will the feature be emitted in IR?
> > 
> > 
> > yes, I was referring `prefer-no-gather`
> 
> Then it's not expected. `prefer-no-gather` is a real tuning feature.

actually, no cpus enabled this tuning feature.

Reply:
> SSELevel relies on feature string in IR attribute as well.
Which one do you mean?